main.go

```diff
@@ -58,61 +64,120 @@ type response struct {
 	TS      string `json:"ts"`
 }
 
-func main() {
-	pflag.Parse()
-
-	token, err := ioutil.ReadFile(tokenFile)
+func sendSlackRequest(token, url string, payload []byte, debug bool) (*http.Response, error) {
+	req, err := http.NewRequest(http.MethodPost, url, bytes.NewBuffer(payload))
 	if err != nil {
-		log.Fatalf("failed to read %s: %s", tokenFile, err)
+		return nil, fmt.Errorf("failed to create request: %w", err)
 	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Authorization", "Bearer "+token)
 
-	payload, err := json.Marshal(&message{
-		Channel:   channel,
-		Text:      text,
-		ThreadTS:  threadTS,
-		LinkNames: true,
-	})
+	client := &http.Client{Timeout: 15 * time.Second}
+	resp, err := client.Do(req)
 	if err != nil {
-		log.Fatalf("failed to marshal message: %s", err)
+		return nil, fmt.Errorf("failed to send request: %w", err)
+	}
+
+	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
+		bodyBytes, _ := io.ReadAll(resp.Body)
+		return nil, fmt.Errorf("request failed: %s", string(bodyBytes))
 	}
 
-	req, err := http.NewRequest(http.MethodPost, "https://slack.com/api/chat.postMessage", bytes.NewBuffer(payload))
+	if debug {
+		bodyBytes, _ := io.ReadAll(resp.Body)
```

Comment:
I suspect that you can't read `resp.Body` twice (with `debug == true` and then later read the response again).